### PR TITLE
completed ticket 11 -> added in query features

### PR DESCRIPTION
**Pull request by zahraakarim**

### CODE REVIEW

File: news.model.js [hunk 1]
```diff
@@ -17,14 +18,34 @@ exports.selectArticleById = (article_id) => {
...
 
-exports.selectAllArticles = () => {
-  return db
-    .query(
-      "SELECT articles.author, articles.article_id, articles.title, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comment_id)::INT AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY created_at DESC;"
-    )
-    .then((response) => {
-      return response.rows;
-    });
+exports.selectAllArticles = (topic, sort_by = "created_at", order = "DESC") => {
```
Comment (johnhutcheon) [inline news.model.js:21]: Nice work giving these defaults 🙌

File: __tests__/app.test.js [hunk 2]
```diff
@@ -114,14 +114,68 @@ describe("GET /api/articles", () => {
...
+      expect(body.msg).toBe("Bad Request");
+    });
+});
+test("404: ERROR: responds with an error when path is valid but does not exist", () => {
+  return request(app)
+    .get("/api/nonesense")
+    .expect(404)
+    .then(({ body }) => {
+      expect(body.msg).toBe("Path not found");
+    });
```
Comment (johnhutcheon) [inline __tests__/app.test.js:178]: Great testing so far! Could we just have two more please? What would we expect to happen in the following cases....

- valid topic query, but has no articles responds with an empty array of articles, e.g. ?topic=paper
- non-existent topic query, e.g. ?topic=bananas

File: __tests__/app.test.js [hunk 3]
```diff
@@ -114,14 +114,68 @@ describe("GET /api/articles", () => {
...
+});
+test("400: ERROR: responds with an error when given an invalid sort_by query", () => {
+  return request(app)
+    .get("/api/articles?sort_by=bobbybrown")
+    .expect(400)
+    .then(({ body }) => {
+      expect(body.msg).toBe("Bad Request");
+    });
+});
+test("404: ERROR: responds with an error when path is valid but does not exist", () => {
```
Comment (johnhutcheon) [inline __tests__/app.test.js]: This test should be in a separate describe block, as it's a generic test for invalid paths, not specific to this ticket 👍

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Zahraa, great work! Could you just amend the testing as per my comments please? Thanks! 😄

File: __tests__/app.test.js (hunk 2 above)
Comment (zahraakarim) [reply __tests__/app.test.js:178]: Did i not do this one? i did ?topic=beans in one of my 200 tests?

**Pull request merged**